### PR TITLE
288-remove

### LINKED ISSUE #288: Add a `remove` method to `LocalStorage`
Several items stored in `LocalStorage` are private, and it would be better if we could remove them from `LocalStorage` when needed, rather than just `retrieve` them.

# Tasks
- [ ] Add `remove` method to `LocalStorage`
- [ ] Write documentation / guidance for its use (e.g., when pulling secrets out of local storage to return them to the user)
- [ ] Call it in all appropriate places across the protocols
- [ ] Remove extra clones that are no longer needed

### PR DESCRIPTION
**Pull request by tyurek**
Adds a new `remove()` function to the Storage API, which removes an item from storage and returns it to the call. Replaces a few end-of-protocol calls to `retrieve` with `remove`, but only where there is no chance that the item will be needed again. Replacing other instances of `retrieve` with `remove` will require more care, especially when considering the possibility of `ProtocolError`s

Closes #288

### CODE REVIEW

File: src/local_storage.rs [hunk 1]
```diff
@@ -93,6 +93,36 @@ impl LocalStorage {
             })
     }
 
+    /// Deletes a storage value via its [`TypeTag`] and
+    /// [`ParticipantIdentifier`]. This function returns the deleted value
+    /// if it existed and should be used instead of retrieve if the value is
```
Comment (marsella) [inline src/local_storage.rs:98]: suggestion: link to retrieve here ```[`retrieve()`](LocalStorage::retrieve())```

**Review by marsella — APPROVED** [review]
Looks good to me! One small note for your consideration.

**Review by marsella — CHANGES_REQUESTED** [review]
Oh sorry, one more thing:

I think you can also remove the `r3_private` and `r3_pubs` in `do_presign_finish`. Is there a reason you skipped those?

**Comment by tyurek** [comment]
After r3_pubs and r3_private are retrieved, there are still checks that need to be done which can fail with ProtocolError. I didn't remove these values, so to avoid figuring out what to do with them in this case

**Review by marsella — APPROVED** [review]
Thanks, I misread the code and thought you could still remove the `private`, but now I see that's not the case.

**Pull request merged**